.github/workflows/release.yaml

```diff
@@ -0,0 +1,73 @@
+name: release
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: "1.15"
+
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-go-
+
+      - uses: actions/cache@v2
+        with:
+          path: ./testbin
+          key: ${{ runner.os }}-testbin
+
+      - uses: actions/cache@v2
+        with:
+          path: ./bin
+          key: ${{ runner.os }}-bin
+
+      - name: Build image
+        run: make docker-build
+
+      - name: Export release name
+        run: |
+          echo "NEW_RELEASE=$(make get-new-release)" >> $GITHUB_ENV
+
+      - name: Login to quay.io/3scale
+        if: ${{ env.NEW_RELEASE != '' }}
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.REGISTRY_USER }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+
+      - name: Push new operator image
+        if: ${{ env.NEW_RELEASE != '' }}
+        run: make docker-push
+
+      - name: Login to quay.io/3scaleops
+        if: ${{ env.NEW_RELEASE != '' }}
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.CATALOG_REGISTRY_USER }}
+          password: ${{ secrets.CATALOG_REGISTRY_PASSWORD }}
+
+      - name: Publish new bundle image in saas-operator catalog
+        if: ${{ env.NEW_RELEASE != '' }}
+        run: make bundle-publish
+
+      - name: Create a new draft-release in github
+        if: ${{ env.NEW_RELEASE != '' }}
+        uses: "marvinpinto/action-automatic-releases@latest"
+        with:
+          repo_token: "${{ secrets.GITHUB_TOKEN }}"
+          automatic_release_tag: "${{ env.NEW_RELEASE }}"
+          title: "${{ env.NEW_RELEASE }}"
+          draft: true
```

Comment:
Creates the release as a draft in case we want to add more info or change the generated description. On the other hand we could do this even if the release is published directly by just editint it, so I'm not sure if creating it as draft adds any value.

Reply:
I think it does, as the release is generated by the CI, having to manually create the actual release in GH afterwards defeats the purpose of the automation. And releases are not that frequent, so starting with a draft that requires some human work to proper document the release have a lot of sense.

Reply:
what is the point as a draft? that maybe later you can finally decide to convert it into release, or delete it before never being a release?

Reply:
Because you usually add some information regarding what the release includes.

Reply:
nice!